### PR TITLE
Fixes #35489 - fix duplicate report upload

### PR DESCRIPTION
**Pull request by bastian-src**
The `upload-salt-reports` script logs already-uploaded reports under _/etc/salt/last_uploaded_. It stores the latest job ID which is equivalent to the execution date. Before uploading new reports, it uses this as a reference to verify that it doesn't re-upload existing reports.

The Salt report upload *reactor* must log the latest report upload in _/etc/salt/last_uploaded_ too such that the `upload-salt-reports` command does not re-upload these reports. 
Compare `upload-salt-reports` [here](https://github.com/theforeman/smart_proxy_salt/blob/master/sbin/upload-salt-reports#L141).

### CODE REVIEW

**Comment by Manisha15** [comment]
Code looks good.

File: salt/report_upload/srv/salt/_runners/foreman_report_upload.py [hunk 1]
```diff
@@ -100,7 +125,10 @@ def now(highstate):
 
     try:
         report = create_report(base64.b64decode(highstate))
+        get_lock()
         upload(report)
+        release_lock()
     except Exception as exc:
+        release_lock()
```
Comment (sbernhard) [inline salt/report_upload/srv/salt/_runners/foreman_report_upload.py:132]: I would put the release_lock in a finally block.

**Comment by sbernhard** [comment]
Generally speaking, why is the cron-job installed if the reactor is used? Do we deliver both at the same time? It would be fine, if the reactor is used (and it should be the default), that the cron job is not installed.

File: salt/report_upload/srv/salt/_runners/foreman_report_upload.py (hunk 1 above)
Comment (bastian-src) [reply salt/report_upload/srv/salt/_runners/foreman_report_upload.py]: Good hint, done.

**Comment by bastian-src** [comment]
If we deliver the reactor, we don't need the `upload-salt-reports` script anymore. That's right. I would remove it in a follow-up PR and we must also remove it in foreman-packaging. Therefore, we should bump the version and make a new release.

**Pull request merged**